src/org/sosy_lab/cpachecker/cpa/smg/SMGPredicateManager.java

```diff
@@ -141,18 +186,46 @@ private BooleanFormula addPredicateToFormula(BooleanFormula pFormula, SymbolicRe
     return result;
   }
 
-  public BooleanFormula getPredicateFormula(PredRelation pRelation) {
-    return getPredicateFormula(pRelation, true);
+  public BooleanFormula getPathPredicateFormula(UnmodifiableSMGState pState) {
+    SMGPredicateRelation pRelation = pState.getPathPredicateRelation();
+    BooleanFormula predicateFormula = getPredicateFormula(pRelation, true);
+    predicateFormula = addExplicitFromState(predicateFormula, pState);
+    return predicateFormula;
   }
 
-  public BooleanFormula getErrorPredicateFormula(PredRelation pErrorPredicate, PredRelation
-      pPathRelation) {
+  public BooleanFormula getErrorPredicateFormula(
+      SMGPredicateRelation pErrorPredicate, UnmodifiableSMGState pState) {
     BooleanFormula errorFormula = getPredicateFormula(pErrorPredicate, false);
-    BooleanFormula pathFormula = getPredicateFormula(pPathRelation, true);
+    BooleanFormula pathFormula = getPathPredicateFormula(pState);
+    pathFormula = addExplicitFromState(pathFormula, pState);
     return fmgr.makeAnd(pathFormula, errorFormula);
   }
 
-  private BooleanFormula getPredicateFormula(PredRelation pRelation, boolean conjunction) {
+  private BooleanFormula addExplicitFromState(
+      BooleanFormula pFormula, UnmodifiableSMGState pState) {
+    BooleanFormula result = pFormula;
+    SMGPredicateRelation errorPredicateRelation = pState.getErrorPredicateRelation();
+    SMGPredicateRelation pathPredicateRelation = pState.getPathPredicateRelation();
+    for (Entry<SMGKnownSymbolicValue, SMGKnownExpValue> expValueEntry :
+        pState.getExplicitValues()) {
+      SMGKnownSymbolicValue symbolicValue = expValueEntry.getKey();
+      if (errorPredicateRelation.hasRelation(symbolicValue)
+          || pathPredicateRelation.hasRelation(symbolicValue)) {
+        SMGKnownExpValue explicitValue = expValueEntry.getValue();
+        BitvectorFormula valueFormula = createdValueFormulas.get(symbolicValue);
+        SMGType symbolicType = valueTypes.get(symbolicValue);
+        valueFormula = cast(valueFormula, symbolicType, symbolicType);
+        BooleanFormula equality =
+            fmgr.makeEqual(
+                valueFormula,
+                efmgr.makeBitvector(symbolicType.getCastedSize(), explicitValue.getValue()));
+        result = fmgr.makeAnd(result, equality);
```

Comment:
Small optimization:
collecting boolean formulas in a list and conjuncting all of them once after the loop might be better for the SMT solver.